src/Init/Data/Int/LemmasAux.lean

```diff
@@ -140,4 +140,39 @@ theorem bmod_bmod_of_dvd {a : Int} {n m : Nat} (hnm : n ∣ m) :
   obtain ⟨k, rfl⟩ := hnm
   simp [Int.mul_assoc]
 
+theorem bmod_eq_of_le_of_lt {x : Int} {y : Nat} (hle : -y ≤ x * 2) (hlt : x * 2 < y) :
+    x.bmod y = x := by
+  simp only [Int.bmod_def]
+  rcases x
+  · rw [Int.emod_eq_of_lt (by simp only [ofNat_eq_coe]; omega) (by omega)]; omega
+  · rw [Int.emod_eq_add_self_emod, Int.emod_eq_of_lt (by omega) (by omega)]; omega
+
+theorem mul_le_mul_self_of_natAbs_le {x y : Int} {s : Nat} (hx : x.natAbs ≤ s) (hy : y.natAbs ≤ s) :
+    x * y ≤ s * s := by
+  rcases s with _|s
+  · simp [show x = 0 by omega]
+  · have := Nat.mul_pos (n := (s + 1)) (m := (s + 1)) (by omega) (by omega)
+    by_cases hx : 0 < x <;> by_cases hy : 0 < y
+    · exact Int.mul_le_mul (by omega) (by omega) (by omega) (by omega)
+    · have : x * y ≤ 0 := Int.mul_nonpos_of_nonneg_of_nonpos (by omega) (by omega); omega
+    · have : x * y ≤ 0 := Int.mul_nonpos_of_nonpos_of_nonneg (by omega) (by omega); omega
+    · have : -x * -y ≤ (s + 1) * (s + 1) := Int.mul_le_mul (by omega) (by omega) (by omega) (by omega)
+      simp [Int.neg_mul_neg] at this
+      norm_cast
```

Comment:
```suggestion
    by_cases hx : 0 < x <;> by_cases hy : 0 < y
    · exact Int.mul_le_mul <;> omega
    · have : x * y ≤ 0 := Int.mul_nonpos_of_nonneg_of_nonpos <;> omega
    · have : x * y ≤ 0 := Int.mul_nonpos_of_nonpos_of_nonneg  <;> omega
    · have : -x * -y ≤ (s + 1) * (s + 1) := Int.mul_le_mul <;> omega
      simp [Int.neg_mul_neg] at this
      norm_cast
```

Also, the `simp` is non-terminal

Reply:
This sadly does not seem to work (the `<;> omega` raises a type mismatch error and leaves goals unsolved)

Reply:
I'd suggest marking the issue as resolved, in that case!